alibi_detect/utils/pytorch/kernels.py

```diff
@@ -20,21 +56,307 @@ def sigma_median(x: torch.Tensor, y: torch.Tensor, dist: torch.Tensor) -> torch.
 
     Returns
     -------
-    The computed bandwidth, `sigma`.
+    The logrithm of the computed bandwidth, `log-sigma`.
     """
     n = min(x.shape[0], y.shape[0])
     n = n if (x[:n] == y[:n]).all() and x.shape == y.shape else 0
     n_median = n + (np.prod(dist.shape) - n) // 2 - 1
     sigma = (.5 * dist.flatten().sort().values[n_median].unsqueeze(dim=-1)) ** .5
-    return sigma
+    return sigma.log()
```

Comment:
What is the reason why this is changed within the function? Can't this just be done afterwards? Because it is used in various places, so would need very careful checking, and the name of the function should become `log_sigma_median` or something similar for clarity instead? The behaviour would also deviate from `alibi_detec.cd.pytorch.context_aware._sigma_median_diag`, which leads to confusion.

Reply:
I aimed to declare all sigma parameters directly as `log-sigma` for the non-negative constraint of sigma. But we can go back to the direct sigma to avoid confusion.